tests/rptest/services/redpanda.py

```diff
@@ -118,6 +118,47 @@ def one_or_many(value):
         return value
 
 
+class ResourceSettings:
+    """
+    Control CPU+memory footprint of Redpanda instances.  Pass one
+    of these into your RedpandaTest constructor if you want to e.g.
+    create low-memory situations.
+    """
+    def __init__(self,
+                 num_cpus=None,
```

Comment:
would you be opposed to using `(self, *, num_cpus=...)` it seemed to provide some real world benefits when refactoring python with the assistance of pyright which could reason between positional and kwargs.

Reply:
good call, I've also been trying to add more type annotations so added them here too.